### PR TITLE
F/OS-1224 Multisig approvers not appearing in past proposals

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Some approvers did not appear as approvers on past proposals after they've been removed from the multisig because the entity for the multisift members and approvers was related. 

This fixes this issue using the solution discussed here: https://www.notion.so/aragonorg/Subgraph-Bug-Approvers-Query-Breaks-Multisig-Subgraph-d783e196ba464595974ba2a9d2dac80a#5ca1bd831bbd4dc191dd8fb0d4360e8f 

Task ID: [OS-1224](https://aragonassociation.atlassian.net/browse/OS-1224)

Playground: https://subgraph.satsuma-prod.com/aragon/osx-test-members-bug-mainnet/version/5/playground

## Type of change

See the framework lifecycle in `packages/contracts/docs/framework-lifecycle` to decide what kind of change this pull request is.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.
- [x] I have updated the Subgraph and added a QA URL to the description of this PR.

[OS-1224]: https://aragonassociation.atlassian.net/browse/OS-1224?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/schema.graphql [hunk 1]
```diff
@@ -888,16 +888,24 @@ type MultisigPlugin implements IPlugin @entity {
 }
 
 type MultisigApprover @entity {
+  """
+  Current Approver of the multisig
+  """
   id: ID! # plugin_address + member_address
   address: String # address as string to facilitate filtering by address on the UI
-  proposals: [MultisigProposalApprover!]! @derivedFrom(field: "approver")
```
Comment (clauBv23) [inline packages/subgraph/schema.graphql:893]: if we remove this field, won't be able to get the proposals by approver. 

I guess it is removed because `MultisigApprover` was removed from  `MultisigProposalApprover` but think shouldn’t lose the ability to get all the proposals of an approver.

File: packages/subgraph/src/packages/multisig/multisig.ts [hunk 2]
```diff
@@ -117,8 +119,9 @@ export function handleApproved(event: Approved): void {
     MultisigProposalApprover.load(approverProposalId);
   if (!approverProposalEntity) {
     approverProposalEntity = new MultisigProposalApprover(approverProposalId);
-    approverProposalEntity.approver = memberEntityId;
     approverProposalEntity.proposal = proposalEntityId;
+    approverProposalEntity.plugin = pluginEntityId;
+    approverProposalEntity.address = generateEntityIdFromAddress(memberAddress);
```
Comment (clauBv23) [inline packages/subgraph/src/packages/multisig/multisig.ts:124]: is this an address or an ID?

**Review by clauBv23 — COMMENTED** [review]
let some comments

**Comment by clauBv23** [comment]
Since there is a branch for removing all plugins, multisig included, [f/split-linting](https://github.com/aragon/osx/tree/f/split-subgraph), not sure if this should be done only in this repo

**Comment by josemarinas** [comment]
> Since there is a branch for removing all plugins, multisig included, [f/split-linting](https://github.com/aragon/osx/tree/f/split-subgraph), not sure if this should be done only in this repo

Yes, this should also be done in the split repo, but i wanted to first get the fix reviewed and then aply it also to the other repo

File: packages/subgraph/tests/multisig/multisig.test.ts [hunk 3]
```diff
@@ -420,8 +424,9 @@ test('Run Multisig (handleMembersRemoved) mappings with mock event', () => {
 
   // checks
   assert.fieldEquals('MultisigApprover', memberId1, 'id', memberId1);
-  assert.notInStore('MultisigApprover', memberId2);
-
+  assert.fieldEquals('MultisigApprover', memberId1, 'isActive', 'true');
+  assert.fieldEquals('MultisigApprover', memberId2, 'id', memberId2);
+  assert.fieldEquals('MultisigApprover', memberId2, 'isActive', 'false');
   clearStore();
 });
```
Comment (jordaniza) [inline packages/subgraph/tests/multisig/multisig.test.ts:431]: Can you add a test for the null pointer exception?

File: packages/subgraph/schema.graphql [hunk 4]
```diff
@@ -918,12 +927,12 @@ type MultisigProposal implements IProposal @entity {
...
-  approvals: Int
+  approvalCount: Int
   approvalReached: Boolean!
   isSignaling: Boolean!
   executed: Boolean!
   executionDate: BigInt
   executionBlockNumber: BigInt
   executionTxHash: Bytes
-  approvers: [MultisigProposalApprover!]! @derivedFrom(field: "proposal")
+  approvals: [MultisigProposalApproval!]! @derivedFrom(field: "proposal")
```
[the remark this answers is not in the document]
Comment (josemarinas) [reply packages/subgraph/schema.graphql:937]: Why, this is inside the `MultisigProposal` and references the proposal field in the `MultiisgProposalApproval`

[the remark this answers is not in the document]
Comment (jordaniza) [reply packages/subgraph/schema.graphql:937]: Sorry didn't expand the code block

File: packages/subgraph/schema.graphql [hunk 5]
```diff
@@ -888,14 +888,23 @@ type MultisigPlugin implements IPlugin @entity {
 }
 
 type MultisigApprover @entity {
+  """
+  Current Approver of the multisig
+  """
   id: ID! # plugin_address + member_address
-  address: String # address as string to facilitate filtering by address on the UI
-  proposals: [MultisigProposalApprover!]! @derivedFrom(field: "approver")
+  address: Bytes!
```
[the remark this answers is not in the document]
Comment (josemarinas) [reply packages/subgraph/schema.graphql:895]: `Bytes` are shown as a string in subgraph and can be used for filtering, all the other addresses in the schema are as Bytes, I dindt see the point of  leaving it as a string

File: packages/subgraph/tests/multisig/multisig.test.ts (hunk 3 above)
Comment (josemarinas) [reply packages/subgraph/tests/multisig/multisig.test.ts:431]: Not sure what kind of test are you refering to. We had a null pointer exception when we removed one member. In this test we are doing exactly that and we check that the member was not removed and instead it's still there but the isActive is set to false. We could add a check for checking that the approvals and the member are still there after removing a member, is that what you meant?

**Review by jordaniza — APPROVED** [review]

File: packages/subgraph/schema.graphql (hunk 5 above)
[the remark this answers is not in the document]
Comment (clauBv23) [reply packages/subgraph/schema.graphql:896]: I see no problem with having them as Bytes. my main concern is that they explicitly say `address as string to facilitate filtering by address on the UI`, not sure how correct that comment is but would like to know. 

This is also being done in other places ([here](https://github.com/aragon/osx/blob/c27df037cfb99ad20d9c23a04f20db8f9eb73379/packages/subgraph/schema.graphql#L694), [here](https://github.com/aragon/osx/blob/c27df037cfb99ad20d9c23a04f20db8f9eb73379/packages/subgraph/schema.graphql#L841), [here](https://github.com/aragon/osx/blob/c27df037cfb99ad20d9c23a04f20db8f9eb73379/packages/subgraph/schema.graphql#L892))

I just want to avoid having to affect the APp team in case the comment is correct and is placed as a string to make their work easier.

[the remark this answers is not in the document]
Comment (josemarinas) [reply packages/subgraph/schema.graphql:896]: Yep, that makes sense, I personally see no difference in behaivour, but lets ask @Rekard0 who is who added the comment in the first place

[the remark this answers is not in the document]
Comment (clauBv23) [reply packages/subgraph/schema.graphql:896]: we checked and there is no reason for having it as a string, we also considered that we should unify all the addresses to bytes.

**Review by clauBv23 — APPROVED** [review]
